homeassistant/components/sensor/teleinfo.py

```diff
@@ -0,0 +1,156 @@
+            icon_template: mdi:flash
+    - platform: template
+        sensors:
+        teleinfo_iinst1:
+            value_template: '{{ states.sensor.edf_teleinfo.
+                attributes["IINST1"] | int }}'
+            unit_of_measurement: 'A'
+            icon_template: mdi:flash
+
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_NAME, EVENT_HOMEASSISTANT_STOP, ATTR_ATTRIBUTION)
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['pyserial-asyncio==0.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_SERIAL_PORT = 'serial_port'
+
+CONF_ATTRIBUTION = "Provided by EDF Teleinfo."
+
+DEFAULT_NAME = "Serial Teleinfo Sensor"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SERIAL_PORT): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the Serial sensor platform."""
+    name = config.get(CONF_NAME)
+    port = config.get(CONF_SERIAL_PORT)
+    sensor = SerialTeleinfoSensor(name, port)
+
+    hass.bus.async_listen_once(
+        EVENT_HOMEASSISTANT_STOP, sensor.stop_serial_read())
+    async_add_entities([sensor], True)
+
+
+class SerialTeleinfoSensor(Entity):
+    """Representation of a Serial sensor."""
+
+    def __init__(self, name, port):
+        """Initialize the Serial sensor."""
+        self._name = name
+        self._port = port
+        self._serial_loop_task = None
+        self._state = None
+        self._attributes = {}
+
+    async def async_added_to_hass(self):
+        """Handle when an entity is about to be added to Home Assistant."""
+        self._serial_loop_task = self.hass.loop.create_task(
+            self.serial_read(self._port, baudrate=1200, bytesize=7,
+                             parity='E', stopbits=1, rtscts=1))
+
+    async def serial_read(self, device, **kwargs):
+        """Process the serial data."""
+        import serial_asyncio
+        _LOGGER.debug(u"Initializing Teleinfo")
+        reader, _ = await serial_asyncio.open_serial_connection(url=device,
+                                                                **kwargs)
+
+        is_over = True
+
+        # First read need to clear the grimlins.
+        line = await reader.readline()
+
+        while True:
+            line = await reader.readline()
+            line = line.decode('ascii').replace('\r', '').replace('\n', '')
+
+            if is_over and ('\x02' in line):
+                is_over = False
+                _LOGGER.debug(" Start Frame")
+                continue
+
+            if (not is_over) and ('\x03' not in line):
+                # Don't use strip() here because the checksum can be ' '.
+                if len(line.split()) == 2:
```

Comment:
This whole if/else could be simplified to: `name, value = line.split()[:2]`

Reply:
sure !